### PR TITLE
[feature](Nereids): pushdown MIN/MAX/SUM through join

### PR DESCRIPTION
**Pull request by jackwener**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jackwener** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.43 seconds
 stream load tsv:          505 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17161128235 Bytes

**Review by starocean999 — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PushdownMinMaxThroughJoin.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    public List<Rule> buildRules() {
+        return ImmutableList.of(
+                logicalAggregate(innerLogicalJoin())
+                        .when(agg -> agg.child().getOtherJoinConjuncts().size() == 0)
+                        .whenNot(agg -> agg.child().children().stream().anyMatch(p -> p instanceof LogicalAggregate))
+                        .when(agg -> agg.getGroupByExpressions().stream().allMatch(e -> e instanceof Slot))
+                        .when(agg -> {
+                            Set<AggregateFunction> funcs = agg.getAggregateFunctions();
+                            return !funcs.isEmpty() && funcs.stream()
+                                    .allMatch(f -> f instanceof Min || f instanceof Max && f.child(0) instanceof Slot);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PushdownMinMaxThroughJoin.java:73]:
```suggestion
                                    .allMatch((f -> f instanceof Min || f instanceof Max) && f.child(0) instanceof Slot);
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PushdownMinMaxThroughJoin.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+                        .toRule(RuleType.PUSHDOWN_MIN_MAX_THROUGH_JOIN),
+                logicalAggregate(logicalProject(innerLogicalJoin()))
+                        .when(agg -> agg.child().isAllSlots())
+                        .when(agg -> agg.child().child().getOtherJoinConjuncts().size() == 0)
+                        .whenNot(agg -> agg.child().children().stream().anyMatch(p -> p instanceof LogicalAggregate))
+                        .when(agg -> agg.getGroupByExpressions().stream().allMatch(e -> e instanceof Slot))
+                        .when(agg -> {
+                            Set<AggregateFunction> funcs = agg.getAggregateFunctions();
+                            return !funcs.isEmpty() && funcs.stream()
+                                    .allMatch(f -> f instanceof Min || f instanceof Max && f.child(0) instanceof Slot);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PushdownMinMaxThroughJoin.java:85]:
```suggestion
                                    .allMatch((f -> f instanceof Min || f instanceof Max) && f.child(0) instanceof Slot);
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PushdownMinMaxThroughJoin.java [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        Set<Slot> leftGroupBy = new HashSet<>();
+        Set<Slot> rightGroupBy = new HashSet<>();
+        agg.getGroupByExpressions().forEach(e -> {
+            Slot slot = (Slot) e;
+            if (leftOutput.contains(slot)) {
+                leftGroupBy.add(slot);
+            } else if (rightOutput.contains(slot)) {
+                rightGroupBy.add(slot);
+            } else {
+                throw new IllegalStateException("Slot " + slot + " not found in join output");
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PushdownMinMaxThroughJoin.java:119]: group by 1 will throw exception here

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PushdownSumThroughJoin.java [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+        Set<Slot> leftGroupBy = new HashSet<>();
+        Set<Slot> rightGroupBy = new HashSet<>();
+        agg.getGroupByExpressions().forEach(e -> {
+            Slot slot = (Slot) e;
+            if (leftOutput.contains(slot)) {
+                leftGroupBy.add(slot);
+            } else if (rightOutput.contains(slot)) {
+                rightGroupBy.add(slot);
+            } else {
+                throw new IllegalStateException("Slot " + slot + " not found in join output");
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PushdownSumThroughJoin.java:124]: group by 1 will throw exception

**Comment by jackwener** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.45 seconds
 stream load tsv:          509 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          69 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17162506728 Bytes

**Review by morrySnow — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**